src/DGtal/kernel/IntegerConverter.h

```diff
@@ -0,0 +1,395 @@
+    static void mpz_set_sll(mpz_t n, long long sll)
+    {
+      mpz_set_si(n, (int)(sll >> 32));     /* n = (int)sll >> 32 */
+      mpz_mul_2exp(n, n, 32 );             /* n <<= 32 */
+      mpz_add_ui(n, n, (unsigned int)sll); /* n += (unsigned int)sll */
+    }
+    
+    /// @param[inout] n the (initialized) big integer to set
+    /// @param[in] ull an unsigned long long integer to assign to \a n.
+    static void mpz_set_ull(mpz_t n, unsigned long long ull)
+    {
+      mpz_set_ui(n, (unsigned int)(ull >> 32)); /* n = (unsigned int)(ull >> 32) */
+      mpz_mul_2exp(n, n, 32);                   /* n <<= 32 */
+      mpz_add_ui(n, n, (unsigned int)ull);      /* n += (unsigned int)ull */
+    }      
+
+    /// Conversion to uint64 is tricky and not native for GMP.
+    /// @param n any number
+    /// @return its uint64 representation.
+    static unsigned long long mpz_get_ull(mpz_t n)
+    {
+      unsigned int lo, hi;
+      mpz_t tmp;
+      mpz_init( tmp );
+      mpz_mod_2exp( tmp, n, 64 );   /* tmp = (lower 64 bits of n) */
+      lo = mpz_get_ui( tmp );       /* lo = tmp & 0xffffffff */ 
+      mpz_div_2exp( tmp, tmp, 32 ); /* tmp >>= 32 */
+      hi = mpz_get_ui( tmp );       /* hi = tmp & 0xffffffff */
+      mpz_clear( tmp );
+      return (((unsigned long long)hi) << 32) + lo;
+    }
+    
+    /// Conversion to int64 is tricky and not native for GMP.
+    /// @param n any number
+    /// @return its int64 representation.
+    static long long mpz_get_sll(mpz_t n)
+    {
+      return (long long)mpz_get_ull(n); /* just use unsigned version */
+    }
+  }
+#endif
+    
+  /// ------------- INTEGER/POINT CONVERSION SERVICES --------------------
+    
+  /// Allows seamless conversion of integral types and lattice
+  /// points, while checking for errors when going from a more
+  /// precise to a less precise type.
+  ///
+  /// Generic version allowing only the identity cast.
+  ///
+  /// @tparam dim static constant of type DGtal::Dimension that
+  /// specifies the static  dimension of the space and thus the number
+  /// of elements  of the Point or Vector.
+  ///
+  /// @tparam TInteger an integral type, a model of concepts::CInteger
+  template < DGtal::Dimension dim,
+             typename TInteger >
+  struct IntegerConverter {
+    BOOST_CONCEPT_ASSERT(( concepts::CInteger<TInteger> ));
+    typedef TInteger Integer;
+
+    /// @param i any integer
+    /// @return the same integer
+    static Integer cast( Integer i ) 
+    {
+      return i;
+    }
+
+    /// Conversion of a lattice point.
+    ///
+    /// @param p any point
+    /// @return the same point
+    static
+    PointVector< dim, Integer >
+    cast( PointVector< dim, Integer > p )
+    {
+      return p;
+    }
+  };
+
+  /// Allows seamless conversion of integral types and lattice
+  /// points, while checking for errors when going from a more
+  /// precise to a less precise type.
+  ///
```

Comment:
For all this, why not relying on NumericalTraits?

Reply:
Maybe this is introducing duplicate codes/features

Reply:
Because they are not sufficient for this purpose.
In fact gmp cannot initialize a big int from an int64 variable (but can from a const expression).
Conversely, gmp can do some conversion with long (which may or may not be 64 bits), but not on long long.
Finally, I wish to print warnings when bad conversions occur.

Reply:
should we remove the NumberTraits converter and use this one?

Reply:
Perhaps we have to make more checks before doing that. Anyway the numbertraits converter just converts to int64. I was afraid that it was incorrect for BigInteger but it was working on my mac (where long is also 64 bits). So perhaps it is no harm keeping it. Otherwise, a cleaner IntegerConverter would have a "status", instead of displaying warnings. We can perhaps make an issue and I do a small PR afterwards to do that. What do you think ? Among other limitations, I only treat the signed case, and I do not care about small integers.

Reply:
I don't know.. I am afraid that having converters at several location could be a problem (like the dozen of functors/classes to threshold an image). For me, NumberTraits is a good place to contain such helpers/converters. Let me check again the converters you have here...